### PR TITLE
Bug killing spree

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Hide `ignore user` caret when user is logged off
- Always default to moderation queue tab `all` 
- Show whole long comments instead of cutting it off
- Prevent double execution when fetching new comments
- Click outside of `ignore user` dialogue closes the dialogue
- Remove `react-onclickoutside` in favor of a custom developed Component called `ClickOutside` inspired by https://github.com/tj/react-click-outside that also supports detecting clicks outside of the iframe!
- Permlink Dialogue, Report Dialogue and the Ignore Dialogue now also hides when a click occurred outside of the iframe!

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -53,7 +53,8 @@
         "no-lonely-if": [2],
         "curly": [2],
         "no-unused-vars": ["error", {
-            "argsIgnorePattern": "next"
+            "argsIgnorePattern": "^_|next",
+            "varsIgnorePattern": "^_"
```
Comment (wyattjoh) [inline .eslintrc.json:57]: I did not know that existed... So cool...

File: client/coral-embed-stream/src/components/Toggleable.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import styles from './Toggleable.css';
+import classnames from 'classnames';
+
+const upArrow = <span className={classnames(styles.chevron, styles.up)}></span>;
+const downArrow = <span className={classnames(styles.chevron, styles.down)}></span>;
+export default class Toggleable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.toggle = this.toggle.bind(this);
+    this.close = this.close.bind(this);
```
Comment (okbel) [inline client/coral-embed-stream/src/components/Toggleable.js]: Why are we re-binding this if we are already doing this `close = () => {` ?

File: client/coral-framework/components/ClickOutside.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import {Component, cloneElement, Children} from 'react';
+import PropTypes from 'prop-types';
```
Comment (okbel) [inline client/coral-framework/components/ClickOutside.js:2]: 👍

File: client/coral-framework/components/ClickOutside.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    pym.onMessage('click', this.handleClick);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleClick, true);
+    pym.messageHandlers.click = pym.messageHandlers.click.filter((h) => h !== this.handleClick);
+  }
+
+  render() {
+    const {children, onClickOutside: _, ...rest} = this.props;
```
Comment (okbel) [inline client/coral-framework/components/ClickOutside.js:32]: Any specific reason why we are renaming `onClickOutside: _` this and not using it? Just curious

File: client/coral-embed-stream/src/components/Toggleable.js (hunk 2 above)
Comment (alinhle) [reply client/coral-embed-stream/src/components/Toggleable.js]: Good point.

File: client/coral-framework/components/ClickOutside.js (hunk 4 above)
Comment (alinhle) [reply client/coral-framework/components/ClickOutside.js:32]: Yes, so it is filtered out of `rest`. I added a rule to `eslint` to ignore any arguments and variables starting with `_`.

**Comment by kgardnr** [comment]
I've confirmed this fixes all the bugs listed above! Ready to merge if @okbel approves :)

**Review by kgardnr — APPROVED** [review]

File: client/coral-embed-stream/src/components/LoadMore.js [hunk 5]
```diff
@@ -23,15 +23,17 @@ class LoadMore extends React.Component {
...
-    const {topLevel, moreComments, loadMore, replyCount} = this.props;
+    const {topLevel, moreComments, replyCount} = this.props;
     return moreComments
       ? <div className='coral-load-more'>
         <Button
-          onClick={() => {
-            this.initialState = false;
-            loadMore();
-          }}>
+          onClick={this.loadMore}>
```
Comment (okbel) [inline client/coral-embed-stream/src/components/LoadMore.js:36]: 👍

File: client/coral-framework/components/ClickOutside.js (hunk 4 above)
Comment (okbel) [reply client/coral-framework/components/ClickOutside.js:32]: I love it. ❤️

**Review by okbel — APPROVED** [review]

**Comment by okbel** [comment]
Looks 💯 😍

**Pull request merged**